### PR TITLE
Make PySpark the only Spark image.

### PR DESCRIPTION
**Pull request by razvan**
# Description

Kuttl tests run locally with https://github.com/stackabletech/spark-k8s-operator/pull/275
Part of https://github.com/stackabletech/spark-k8s-operator/issues/273

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
- [x] Changes are OpenShift compatible
- [x] All added packages (via microdnf or otherwise) have a comment on why they are added
- [x] Things not downloaded from Red Hat repositories should be mirrored in the Stackable repository and downloaded from there
- [x] All packages should have (if available) signatures/hashes verified
- [x] Does your change affect an SBOM? Make sure to update all SBOMs
```

### CODE REVIEW

File: spark-k8s/CHANGELOG.md [hunk 1]
```diff
@@ -18,8 +18,9 @@
 
 ### Changed
 
-- Base image changed to `stackable/image/vector` ([#342]).
```
Comment (lfrancke) [inline spark-k8s/CHANGELOG.md:21]: I'm only on my phone and might misread the diff but why did you remove this? The reference still seems to be there

Comment (razvan) [reply spark-k8s/CHANGELOG.md:21]: It's because the folder has been renamed from pyspark to spark.
But in fact, I would like to get rid of the individual changelog files because:
* they are not maintained 
* they cannot be automatically updated in the release process

Instead, all changes in this repo should be documented in the root changelog file.

Comment (razvan) [reply spark-k8s/CHANGELOG.md:21]: I updated the changelog with the old spark entries so nothing is lost for now: https://github.com/stackabletech/docker-images/pull/436/commits/87b2063f794aa4d231a4c8d2d15184576d3ae8e8

**Review by maltesander — APPROVED** [review]
LGTM!

**Review by maltesander — APPROVED** [review]
LGTM!

**Comment by lfrancke** [comment]
Do we need to change documentation for this?
Do we need a changelog entry for this?
Migration steps?

**Comment by razvan** [comment]
The documentation has been updated in the operator.
Good question regarding migration steps: in the next release `SparkApplications` need to be updated anyway because of crd changes. Perhaps add them to the release notes for 23.11 ?
